pkg/proxy/winkernel/proxier.go

```diff
@@ -354,7 +354,7 @@ func newSourceVIP(hns HostNetworkService, network string, ip string, mac string,
 
 func (ep *endpointsInfo) Cleanup() {
 	Log(ep, "Endpoint Cleanup", 3)
-	if ep.refCount != nil {
+	if !ep.GetIsLocal() && ep.refCount != nil {
 		*ep.refCount--
 
 		// Remove the remote hns endpoint, if no service is referring it
```

Comment:
Having trouble with the git UI, so this comment is really for line 363, but I think the !ep.GetIsLocal() check there is now redundant

Reply:
also having more trouble with the UI, but I am wondering why 'ep.refCount = nil' is not within that if block on lines 363-371. If we don't attempt to delete a remote endpoint because the refCount is above zero, why set the refCount to nil? This may be outside the scope of this change, but just doesn't make sense to me

Reply:
Consider two services with endpoint objects:

Service1 -> Endpoint1, Endpoint2, Endpoint3

Service2 -> Endpoint1, Endpoint2

And, Service1_Endpoint1 and Service2_Endpoint2 are referring to the same HNS remote endpoint

Original behavior:

Service1_Endpoint1 -> HNSRemoteEndpoint1RefCount
							  
Service1_Endpoint2 -> HNSRemoteEndpoint1RefCount

Deleting Service1 would drop (Service1_Endpoint1 -> HNSRemoteEndpoint1RefCount) to zero and corresponding HNS remote endpoint was deleted, though Service 2 was still using it 


Modified behavior:

Service1_Endpoint1 -> *HNSRemoteEndpoint1RefCount ------>
							                                                 Shared_HNSRemoteEndpoint1RefCount
Service1_Endpoint2 -> *HNSRemoteEndpoint1RefCount ------>

HNSRemoteEndpoint1RefCount is a shared counter and actual refcount's in the endpoint objects are pointers to this shared refcount. Everytime endpoint for a service is cleaned shared refcount is decremented and the refcount pointer on the endpoint is set to nil, so that it can no longer accidentally modify the shared refcount. Once the shared refcount drops to 0 (i.e. no service is using it anylonger) we delete the HNSRemoteEndpoint